aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -129,16 +130,29 @@ fn native_try_add(
     mut args: VecDeque<Value>,
 ) -> SafeNativeResult<SmallVec<[Value; 1]>> {
     debug_assert_eq!(args.len(), 2);
-
     context.charge(AGGREGATOR_V2_TRY_ADD_BASE)?;
-    let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
-    let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
-    let value = pop_value_by_type(&ty_args[0], &mut args)?;
-    let (id, limit) = pop_aggregator_fields_by_type(&ty_args[0], &mut args)?;
-    let aggregator = aggregator_data.get_aggregator(id, limit)?;
-    Ok(smallvec![Value::bool(
-        aggregator.try_add(aggregator_context.resolver, value)?
-    )])
+
+    let input = pop_value_by_type(&ty_args[0], &mut args)?;
+    let agg_struct = safely_pop_arg!(args, StructRef);
+    let (agg_value, agg_limit) = get_aggregator_fields_u128(&agg_struct)?;
+
+    let result_value = if context.aggregator_execution_enabled() {
+        let id = aggregator_value_field_as_id(agg_value)?;
+        let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
+        let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
+        let aggregator = aggregator_data.get_aggregator(id, agg_limit)?;
+        aggregator.try_add(aggregator_context.resolver, input)?
+    } else {
+        match addition(agg_value, input, agg_limit) {
```

Comment:
I think you can use and_then here
```
addition(agg_value, input, agg_limit).and_then(|sum| set_aggregator_value_field(&agg_struct, create_value_by_type(&ty_args[0], sum)?)?).is_ok()?
```

Reply:
I need to propagate errors from inside, while return false on the addition error - so I don't think your change is correct.